engine/gamesys/src/gamesys/test/material/material_constant_array_v140.fp

```diff
@@ -0,0 +1,29 @@
+#version 140
+
+out vec4 FragColor;
+
+struct nested_nested_type
+{
+    vec4 tint[2];
+    vec4 leaf_value[2];
+    vec4 overridden_value[2];
```

Comment:
Note: We can only support arrays for _leaf_ nodes in a nested type chain. This is because we don't have a way of addressing array elements earlier, e.g:

```
struct MyElement {
  vec4 data1;
  vec4 data2;
};

uniform MyUniforms {
  MyElement elements[4];  
};
```

We can't address alements[1].data1 for example, using go.set or a constant buffer.